palantirnet/drupal-10-development
# DEV-39: Add "origin" option to ddev clone command

[DEV-39: Add origin path option to ddev clone command in drupal-10-development](https://palantir.atlassian.net/browse/DEV-39)

- Adds the ability to set the repository origin when using `ddev clone`
- Moves the command from .ddev/commands/web to .ddev/commands/host, so it runs on the host (where Git is usually properly configured) instead of inside the web Docker container.

## Testing

- `ddev clone some_private_module -o git@github.com:palantirnet`
- Verify the module is cloned properly.
- Try the `-b` and `-t` options.
- `ddev clone workbench` (or any module on d.o)
- Verify the module is cloned properly.
- Try the `-b`, `-t`, and `-c` options, and verify they work properly.

## Open questions
- Is it ok to silently ignore `-c` if `-o` is used?
- Is "origin" the right label for this option? It's not a full origin in the Git sense, but it represents the protocol, host, and account portions of Git URLs.
